src/postgres/postgresConnectionStrings.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ConnectionOptions, parse } from "pg-connection-string";
+import { ParsedConnectionString } from "../ParsedConnectionString";
+import { nonNullProp } from "../utils/nonNull";
+
+export async function parsePostgresConnectionString(connectionString: string): Promise<ParsedPostgresConnectionString> {
+    connectionString = connectionString ? connectionString.trim() : '';
```

Comment:
Can `connectionString` be `undefined`? If yes - you should update the type to reflect that. If no - I feel like you could just do `connectionString = connectionString.trim();`